tools/flaky_test_monitor/level2/process_summary2_results.go

```diff
@@ -13,96 +13,107 @@ import (
 const failuresDir = "./failures/"
 const noResultsDir = "./no-results/"
 
-// process level 1 summary files in a single directory and output level 2 summary
-func processSummary2TestRun(level1Directory string) common.TestSummary2 {
-	dirEntries, err := os.ReadDir(filepath.Join(level1Directory))
-	common.AssertNoError(err, "error reading level 1 directory")
-
+func generateLevel2SummaryFromStructs(level1Summaries []common.Level1Summary) common.Level2Summary {
 	// create directory to store failure messages
-	err = os.Mkdir(failuresDir, 0755)
+	err := os.Mkdir(failuresDir, 0755)
 	common.AssertNoError(err, "error creating failures directory")
 
 	// create directory to store no-results messages
 	err = os.Mkdir(noResultsDir, 0755)
 	common.AssertNoError(err, "error creating no-results directory")
 
-	testSummary2 := common.TestSummary2{}
-	testSummary2.TestResults = make(map[string]*common.TestResultSummary)
+	level2Summary := common.Level2Summary{}
+	level2Summary.TestResultsMap = make(map[string]*common.Level2TestResult)
+
+	// go through all level 1 test runs create level 2 summary
+	for i := 0; i < len(level1Summaries); i++ {
+		for _, level1TestResultRow := range level1Summaries[i].Rows {
+			// check if already started collecting summary for this test
+			level2TestResultsMapKey := level1TestResultRow.TestResult.Package + "/" + level1TestResultRow.TestResult.Test
+			level2TestResult, level2TestResultExists := level2Summary.TestResultsMap[level2TestResultsMapKey]
+
+			// this test doesn't have a summary so create it
+			// no need to specify other fields explicitly - default values will suffice
+			if !level2TestResultExists {
+				level2TestResult = &common.Level2TestResult{
+					Test:    level1TestResultRow.TestResult.Test,
+					Package: level1TestResultRow.TestResult.Package,
+				}
+			}
+			// keep track of each duration so can later calculate average duration
+			level2TestResult.Durations = append(level2TestResult.Durations, level1TestResultRow.TestResult.Elapsed)
+
+			// increment # of passes, fails, skips or no-results for this test
+			level2TestResult.Runs++
+			switch level1TestResultRow.TestResult.Result {
+			case "1":
+				level2TestResult.Passed++
+			case "0":
+				level2TestResult.Failed++
+				saveFailureMessage(level1TestResultRow.TestResult)
+			case "skip":
+				level2TestResult.Skipped++
+
+				// don't count skip as a run
+				level2TestResult.Runs--
+
+				// truncate last duration - don't count durations of skips
+				level2TestResult.Durations = level2TestResult.Durations[:len(level2TestResult.Durations)-1]
+			case "-100":
```

Comment:
is -100 just a random magic value we chose to represent no result?

Reply:
Yes, it's explained in the Level 1 parser in more detail so I copied the comments from there to here so it's more clear:

https://github.com/onflow/flow-go/pull/2102/commits/424731838609dbd97fea06074b1afb62c1edfad8

```
// for tests that don't have a result generated (e.g. using fmt.Printf() with no newline in a test)
// we want to highlight these tests so they show up at the top in Granfa
// we do this by simulating a really low fail rate so that the average success rate
// will be very low compared to other tests so it will show up on the "flakiest test" panel
```

Reply:
removed using magic value and treating no-results as a failure but with additional bool field set to true

https://github.com/onflow/flow-go/pull/2102/commits/efcb8f09b5efd63a963b4eb949ac8c793af1eec2